internal/winapi/process.go

```diff
@@ -1,10 +1,80 @@
 package winapi
 
+import (
+	"errors"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
 const PROCESS_ALL_ACCESS uint32 = 2097151
 
-// DWORD GetProcessImageFileNameW(
-//	HANDLE hProcess,
-//	LPWSTR lpImageFileName,
-//	DWORD  nSize
+const (
+	PROC_THREAD_ATTRIBUTE_PSEUDOCONSOLE = 0x20016
+	PROC_THREAD_ATTRIBUTE_JOB_LIST      = 0x2000D
+)
+
+type ProcThreadAttributeList struct {
+	_ [1]byte
+}
+
+// typedef struct _STARTUPINFOEXW {
+// 	STARTUPINFOW                 StartupInfo;
+// 	LPPROC_THREAD_ATTRIBUTE_LIST lpAttributeList;
+// } STARTUPINFOEXW, *LPSTARTUPINFOEXW;
+type StartupInfoEx struct {
+	// This is a recreation of the same binding from the stdlib. The x/sys/windows variant for whatever reason
```

Comment:
This is kind of concerning. Can you share some sample code exhibiting the problem? I can help take a look at what might be failing.

Reply:
Yea.. This seems to be a common issue on the pseudo console packages I'd looked into. They report almost 50% of the time it happening (none of them are using x/sys/windows' ProcThreadAttribute bindings however, they're rolling their own). I ran a test of this in a loop for 10000 iterations and never hit it, so it seems to like the stdlibs variant, which this code is just a copy of. 

0xc0000142 occurs when using the pseudo console api if the handle is invalid or the pseudo console session is closed while starting up and connecting (https://docs.microsoft.com/en-us/windows/console/creating-a-pseudoconsole-session#creating-the-hosted-process). I can't imagine the list is getting freed from looking at x/sys/windows' implementation or anything else of the sort.  https://cs.opensource.google/go/x/sys/+/0c823b97:windows/exec_windows.go;l=141

Reply:
I'll paste some code shortly, I need to recreate what I had for it.

Reply:
Am I understanding right that other people doing pseudo console stuff in Go experience a 50% failure rate when creating processes with a pseudo console associated, but we haven't seen similar failures with the implementation in this PR? Do we know what is different between the implementations?

Are you referring to the `x/sys/windows` code as "the stdlibs variant", or is that something else?

Reply:
@kevpar Correct. The x/sys/windows has a bit of a different definition for the attribute list, initializing it and updating it. The stdlibs seems to work fine. Notably is x/sys/windows went the approach of using LocalAlloc for updating the list 

1. [stdlib](https://github.com/golang/go/blob/a52e4b9c7e8f5aae678596e0c198e67b3b2b1087/src/syscall/syscall_windows.go#L1287)
2. [x/sys/windows](https://cs.opensource.google/go/x/sys/+/0c823b97:windows/exec_windows.go;l=141) 

Here's some examples of people mentioning this bug:
1. https://github.com/marcomorain/go-conpty (the README)
2. https://github.com/creack/pty/pull/109 (PR description)

Reply:
Okay, by stdlib you mean `syscall` package. That's what was missing.

Reply:
Is this resolved? I'm also a bit concerned by needed to recreate this

Reply:
In one way or another we need a binding for it, the way they've went in the stdlib seems to be the correct one for now and was what I would've envisioned in the first place. Would be nice to spend a day or two to fiddle in windbg to see where the culprit lies with the x/sys/windows approach but this works.

Reply:
I'll gladly look into why the x/sys/windows binding is screwed, but I don't think it should halt getting this in.

Reply:
I'll work with the console folks to find some nice bps to place to debug this. I also found a rust issue of someone trying out binds for this and hitting the same error (I'll need to dig up that issue again), makes me think there may be something wonky going on in pty land..